src/peft/config.py

```diff
@@ -213,6 +214,21 @@ def _get_peft_type(
         loaded_attributes = cls.from_json_file(config_file)
         return loaded_attributes["peft_type"]
 
+    @classmethod
+    def check_kwargs(cls, **kwargs):
+        r"""
+        Check if the kwargs are valid for the configuration.
+
+        Args:
+            kwargs (additional keyword arguments, *optional*):
+                Additional keyword arguments passed along to the child class initialization.
+        """
+        if "oft_block_size" in kwargs:
+            warnings.warn(
+                'OFT has been updated since 0.12.1.dev0. Your trained adapter weights may not be compatible with the latest version of OFT. Please retrain your adapter weights.')
```

Comment:
1. Note that `dev0` is not a real version, so let's not refer to that. The next release version will be 0.13.0, so let' use that version.
2. "may not be compatible": We are pretty sure it is incompatible when trained, right? Let's phrase it as "is incompatible".
3. Let's also mention that users can downgrade PEFT to version 0.12.0 and then the adapter will still work.